src/box/sql/vdbe.c

```diff
@@ -1988,6 +1987,18 @@ case OP_Fetch: {
 			diag_set(ClientError, ER_SQL_CANT_RESOLVE_FIELD, name);
 			goto abort_due_to_error;
 		}
+	} else {
+		/*
+		 * If no format is specified, we assume that vdbe_field_ref
+		 * contains a single field for a field constraint.
+		 */
+		if (p->arg_count == ARG_COUNT_MANY) {
+			diag_set(ClientError, ER_SQL_EXECUTE,
+				 "Number of arguments in a field constraint "
+				 "function is greater than one");
+			goto abort_due_to_error;
+		}
+		id = 0;
```

Comment:
You should be able to validate that at attempt to create such field constraint. See my previous comment. (In case function body can't be altered.)

Reply:
Done.